File: 2110-number-of-smooth-descent-periods-of-a-stock/2110-number-of-smooth-descent-periods-of-a-stock.cpp
class Solution {
public:
    long long getDescentPeriods(vector<int>& prices) {
        int n = prices.size();
        if (n == 1) return 1;
        
        long long descent_days = 0;
        
        int i = 0, j = 0;
        while (j < n) {
            i = j;
            while (j < n - 1 && prices[j] - prices[j + 1] == 1) j++;
            
            int length = j - i + 1;
            descent_days += (((long long)length * (length + 1)) / 2);
            j++;
        }
        
        return descent_days;
    }
};

File: 2110-number-of-smooth-descent-periods-of-a-stock/2110-number-of-smooth-descent-periods-of-a-stock.java
class Solution {
    public long getDescentPeriods(int[] prices) {
        long curr = 0;
        long descent_days = 0;
        
        for (int i = 0; i < prices.length; i++) {
            if (i != 0 && prices[i - 1] - prices[i] == 1) curr++;
            else curr = 1;
            
            descent_days += curr;
        }
        
        return descent_days;
    }
}